test/Container/ApplicationFactoryTest.php

```diff
@@ -27,6 +27,68 @@ public function setUp()
         $this->factory   = new ApplicationFactory();
     }
 
+    protected function getContainer($config = null, $router = null, $finalHandler = null, $emitter = null)
```

Comment:
But this would allow us to have only one place for managing and maintaining container configuration and it also solves code duplication in all the test cases in this class. I agree though that it is somewhat unreliable due to its complexity, but after all, every test case which consumes will fail if it does not work as expected.